tutorials/23_Answering_Multihop_Questions_with_Agents.ipynb

```diff
@@ -0,0 +1,5691 @@
+        "\n",
+        "On the other hand, it does well with a question such as \"Who was the 44th president of the USA?\""
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 930,
+          "referenced_widgets": [
+            "d31a483cf6e443a5891ca14672ffed3b",
+            "a0dcb1d860704bbdb54447dfd0cc92bd",
+            "5e85990cb7374ddc9f7107a72f812e5e",
+            "acf1cdc5c6484d8ab5f78ac217656b17",
+            "083cb8b3aa1240129a75e5c2a7eacbb6",
+            "7f4d17a4998f44b58f813b815d16f7f8",
+            "6ffba94cc23846209c4629cb10bce042",
+            "de52fc90bd3f4db0b540535246b33c4f",
+            "acd5844ab2d64071a1d59a1df7baafbd",
+            "744746530a4e401bb6e9e28d89e54eed",
+            "3bb3e922fc8c471a8c1aaaf5bbca2d92"
+          ]
+        },
+        "id": "HdS-satXbSsw",
+        "outputId": "751e30c8-50ae-48d3-8e34-df7023d1252a"
+      },
+      "outputs": [],
+      "source": [
+        "from haystack.utils import print_answers\n",
+        "\n",
+        "#result = presidents_qa.run(\"Who was the 44th president of the USA?\")\n",
+        "result = presidents_qa.run(\"Who is the wife of the 44th president of the USA?\")\n",
+        "\n",
+        "print_answers(result, \"minimum\")"
+      ]
+    },
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bn83I_Dg8Q1f"
+      },
+      "source": [
+        "## Create an Agent with the `ExtractiveQAPipeline` as a `Tool`\n",
+        "### 1) To create the Agent, we'll make use of an Open AI model. So first, provide your Open AI key:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "kBkMr86vvl9h",
+        "outputId": "8366a06d-9064-485e-aa88-c82859c68265"
+      },
+      "outputs": [],
+      "source": [
+        "from getpass import getpass\n",
+        "\n",
+        "api_key_prompt = \"Enter OpenAI API key:\" \n",
+        "api_key = getpass(api_key_prompt)"
+      ]
+    },
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "qajsCzjx9C6j"
+      },
+      "source": [
+        "### 2) Initialize the Agent \n",
+        "\n",
+        "The `Agent` needs to determine the next best course of action at each iteration. It does this by using an LLM, and a prompt designed specially for this use case. Our `Agent` does this by using a `PromptNode` which by default uses the [\"zero-shot-react\" `PromptTemplate` ](https://github.com/deepset-ai/haystack/blob/444a3116c42d2c8852d27aa8093ac92c8e85ab88/haystack/nodes/prompt/prompt_node.py#L337). \n",
```

Comment:
"Our Agent does this" -> Our Agent uses a PromptNode with the default zero-shot-react... (to avoid repeating "does this")